client/my-sites/checkout/composite-checkout/components/prepurchase-notices/index.jsx

```diff
@@ -72,35 +71,41 @@ const PrePurchaseNotices = () => {
 		return products.filter( ( p ) => ! p.expired );
 	} );
 
-	/**
-	 * The active product on the current site that overlaps/conflicts with the plan currently in the cart.
-	 */
-	const siteProductThatOverlapsCartPlan = useSelector( ( state ) => {
+	const siteProductThatOverlapsCartPlan = useSelector( () => {
```

Comment:
Since we're not using the `state` anymore, I believe we can get rid of `useSelector`, and maybe use `useCallback` if we want to avoid re-creating the function on each render.

Reply:
fixed

Reply:
or not....  So I found when I'm not doing useSelector, it triggers the notice without a product (even with the logic fixed on the Security plan) I've tried useMemo and useCallback and nothing at all, nothing passes the correct overlapping product to the notification.  I might mean that I need to pass that some other way by adjusting the prepurchasenotice component itself but that doesn't feel right.  Can you provide some additional insight to the right approach?

Reply:
>  I've tried useMemo and useCallback and nothing at all, nothing passes the correct overlapping product to the notification.

Just to clear that out, did you pass `currentSiteProducts` and  `planSlugInCart` as depencies to `useCallback`?

Reply:
Yeah, but I forgot that useCallback turns that into a function, so I had to 

```
if ( siteProductThatOverlapsCartPlan ) {
		return (
			<CartPlanOverlapsOwnedProductNotice
				product={ siteProductThatOverlapsCartPlan() }
				selectedSite={ selectedSite }
			/>
		);
	}
```
	
Now it's all happy.

Reply:
@TheSteveK In general, the rule of thumb I use for determining when to use `useCallback` and when to use `useMemo` is whether or not I'll need to supply an argument that I'll only know later. For example, if I have a variable that'll need to pull a value from an input field, I can create a callback that'll take that value as a parameter. For data I can figure out with no outside intervention, `useMemo` is easier and more performant (fun fact: function calls actually cost a little more processor time).

In your case, I'd suggest converting to `useMemo` so that `siteProductThatOverlapsCartPlan` becomes a variable and eliminates an extra function call.

Reply:
You're right, for some reason I thought I had unsuccessfully tried that, but it works just fine.